docs/ovs_communication.adoc

```diff
@@ -65,9 +85,31 @@ TBD - XiaoDong
 
 . when and who will initialize OVS? What are the initialization steps
 .. br-int, br-tun are created during init time, and recreated later in RPC loop if needed
-.. need to determine the minimal initialization steps
-. what happen when there are multiple physical NICs on the system, which NIC do we pick to hook up to br-tun, br-vlan, br-ex (if needed)?
+A: for the 1st questions:
+.. there are several ways to init ovs, one is start openvswith service directly or we could do the following
+a. start ovsdb-server, b. start vswitchd, use cmd to init as the following cmd:
+
+a. ovsdb-server --remote=punix:/usr/local/var/run/openvswitch/db.sock  \
+  --remote=db:Open_vSwitch,Open_vSwitch,manager_options \
+  --private-key=db:Open_vSwitch,SSL,private_key  \
+  --certificate=db:Open_vSwitch,SSL,certificate     \
+  --bootstrap-ca-cert=db:Open_vSwitch,SSL,ca_cert  \
+  --log-file=/var/log/openvswitch/ovs-vswitchd.log \
+  -vsyslog:dbg -vfile:dbg  --pidfile --detach
+ 
+b. ovs-vswitchd -v --pidfile --detach \
+ --log-file=/var/log/openvswitch/ovs-vswitchd.log \
+ -vconsole:err -vsyslog:info -vfile:info
+c. ovs-vsctl --no-wait init
+
+for the 2nd questions:
+a. create bridge (adding internal port)
+b. create tunnel to the bridge
+
+. what happen when there are multiple physical NICs on the system, which NIC do we pick to hook up to br-tun, br-vlan, br-ex (if needed)? 
+A:IMHO, during creation of ovs bridge there would be one virtual nic created with same name to the bridge , and if you run your compute instance inside container, there would be one more port on the host virtual nic which would be used as the hookup ( actually it's a pair veth device, one side conneted to vnic)
 .. what is the exact command to connect the new port to the new br-tun, and enable encap/decap?
+A:when you say port, i would assume you mean port in ovs instead of the OS socket port. In that case, if you only connect the port to the bridge , in ovs-vsctl cmd, it would be ovs-vsctl add-port bridgeName (br-tun name) portName
```

Comment:
yes, that's what I meant about port. It is created by Nova and connect to OVS bridge. What do we need to do for configuration this new network port so that I will work for VMs/Containers?

Reply:
please see below example:
     ip netns add nsvm1
     ip link add tap1 type veth peer name tap2
     ip link set tap1 netns nsvm1
     ip netns exec nsvm1 ip addr add 192.168.0.11/24 dev tap1
     ip netns exec nsvm1 ip link set tap1 up
    ip link set tap1 up
    ovs-vsctl add-port br0 tap1

Reply:
Please see the example below:
    ip netns add nsvm2
    ip link add tap3 type veth peer name tap4
    ip link set tap3 netns nsvm2
    ip netns exec nsvm2 ip addr add 192.168.0.12/24 dev tap3
    ip netns exec nsvm2 ip link set tap3 up
    ip link set tap4 up
    ovs-vsctl add-port br0 tap2